rocketpy/monte_carlo/mc_flight.py

```diff
@@ -1,90 +1,140 @@
+    ):
+        """Initializes the Monte Carlo Flight class.
 
-    def rnd_inclination(self):
-        """Creates a random inclination
+        See Also
+        --------
+        This should link to somewhere that explains how inputs works in
+        dispersion models.
 
-        Returns
-        -------
-        inclination : float
-            Random inclination
+        Parameters
+        ----------
+        flight : Flight
+            The Flight object to be used as a base for the Monte Carlo flight.
+        rail_length : int, float, tuple, list, optional
+            The rail length of the flight. Follows the standard input format of
+            Dispersion Models.
+        inclination : int, float, tuple, list, optional
+            The inclination of the flight. Follows the standard input format of
+            Dispersion Models.
+        heading : int, float, tuple, list, optional
+            The heading of the flight. Follows the standard input format of
+            Dispersion Models.
+        initial_solution : tuple, list, optional
+            The initial solution of the flight. This is a tuple of 14 elements
+            that represent the initial conditions of the flight. This attribute
+            can not be randomized.
+        terminate_on_apogee : bool, optional
+            Whether or not the flight should terminate on apogee. This attribute
+            can not be randomized.
         """
-        gen_dict = next(self.dict_generator())
-        return gen_dict["inclination"]
+        if terminate_on_apogee is not None:
+            assert isinstance(
+                terminate_on_apogee, bool
+            ), "`terminate_on_apogee` must be a boolean"
+        super().__init__(
+            flight,
+            rail_length=rail_length,
+            inclination=inclination,
+            heading=heading,
+        )
 
-    def rnd_heading(self):
-        """Creates a random heading
+        self.initial_solution = initial_solution
+        self.terminate_on_apogee = terminate_on_apogee
 
-        Returns
-        -------
-        heading : float
-            Random heading
+    def _validate_initial_solution(self, initial_solution):
+        if initial_solution is not None:
+            if isinstance(initial_solution, (tuple, list)):
+                assert len(initial_solution) == 14, (
+                    "`initial_solution` must be a 14 element tuple, the "
+                    "elements are:\n t_initial, x_init, y_init, z_init, "
+                    "vx_init, vy_init, vz_init, e0_init, e1_init, e2_init, "
+                    "e3_init, w1Init, w2Init, w3Init"
+                )
+                assert all(
+                    [isinstance(i, (int, float)) for i in initial_solution]
+                ), "`initial_solution` must be a tuple of numbers"
+            else:
+                raise TypeError("`initial_solution` must be a tuple of numbers")
+
+    # TODO: these call dict_generator a lot of times unecessaryly
```

Comment:
why is that happening?

Reply:
> This is necessary in the `Dispersion` class to run a simulation.
> 
> To define a `McFlight` you need to pass a `Flight` object, which has a `Rocket` and a `Environment`. However, the `Dispersion` class receives a `McFlight`, `McRocket` and `McEnvironment` and has to construct a brand new flight from those objects. The `McFlight` `create_object` method is never used in the Dispersion class, instead you only need randomized values for rail_length, inclination and heading
> 
> This PR doesn't change anything in the `Dispersion` class, so I simply converted the `McFlight` class to not have Pydantic, but keep it working the same as before.
> 
> Whenever we start working on the `Dispersion` class we should discuss this further. Honestly, I do not think that the `McFlight` class should exist, or at least it should not be passed to the `Dispersion` class

We can re-discuss in the future, the whole architecture of Monte Carlo analysis is something we can optimize before merging to develop.